### ISSUE TITLE
Clamping of consecutive elements with a common parent is broken

### ISSUE THREAD

**Issue by codener** [makandra/superclamp#2]
Consider this markup:
```
<div class="common-parent">
  <div class="first" clamp>Some ... text</div>
  <div class="second" clamp>Some ... text</div>
</div>
```
In this situation, Superclamp will not always clamp the `.second` element properly, i.e. its contents will exceed its height and bleed out of `.second`. Unfortunately, this behavior is not deterministic, meaning it will sometimes clamp right and sometims do it wrong.

When investigating a broken `.second`, I found that Superclamp had stored incorrect `elementAt` values for the element.

I assume this originates from the fact that Superclamp splits "measurement" and "adaption" in two consecutive phases. This breaks when the measures for `.second` are changed during the adaption of `.first`.

**Comment by makmic** [comment]
I managed to create a minimal example, where the second of two subsequent elements is not clamped correctly.

https://github.com/makandra/superclamp/commit/c646b6c131d423ca1bf08e362ddfdf67a93de9e6
![image](https://user-images.githubusercontent.com/32639833/67467874-e665fd80-f649-11e9-8e44-424e276e82b3.png)


As this error only occurs when using `max-height`, my naive guess would be to process clamping jobs ordered via LIFO, that the second element is always processed before the first. ([Code](https://github.com/makandra/superclamp/blob/master/src/superclamp.coffee#L218))

```diff
drainPhaseQueue = (phase) ->
  jobs = jobQueues[phase]
  if jobs.length > 0
-    while job = jobs.shift()
+    while job = jobs.pop()
      job()
```

Sadly, this does not fix the issue. Any other ideas?

**Comment by makmic** [comment]
One more hint: Disabling value-caching fixes the problem, but is not an option:

```diff
- instance = element[INSTANCE_KEY] || new Superclamp(element)
+ instance = new Superclamp(element)
```

**Comment by foobear** [comment]
Fixed in 0.2.3
